nsv13/code/modules/overmap/fighters/fighters_components.dm

```diff
@@ -285,4 +285,242 @@
 	desc = "A tank full of high performance engine fuel, spiked with a little plasma for an extra kick."
 	icon_state = "tank_stationairy" //temp
 	reagent_id = /datum/reagent/plasma_spiked_fuel
-	tank_volume = 2500
\ No newline at end of file
+	tank_volume = 2500
+
+/datum/chemical_reaction/plasma_spiked_fuel
+	name = "Plasma Spiked Fuel"
+	id = /datum/reagent/plasma_spiked_fuel
+	results = list(/datum/reagent/plasma_spiked_fuel = 5)
+	required_reagents = list(/datum/reagent/fuel = 5, /datum/reagent/stable_plasma = 1, /datum/reagent/hydrogen = 2, /datum/reagent/oxygen = 1)
+	required_temp = 333
+
+/datum/chemical_reaction/aviation_fuel
+	name = "Aviation Fuel"
+	id = /datum/reagent/aviation_fuel
+	results = list(/datum/reagent/aviation_fuel = 5)
+	required_reagents = list(/datum/reagent/fuel = 1, /datum/reagent/hydrocarbon=5)
+	required_temp = 333
+
+/datum/chemical_reaction/hydrocarbon
+	name = "Hydrocarbon"
+	id = /datum/reagent/hydrocarbon
+	results = list(/datum/reagent/hydrocarbon = 5)
+	required_reagents = list(/datum/reagent/hydrogen = 1, /datum/reagent/carbon = 1)
+	required_temp = 333
+
+/datum/reagent/hydrocarbon
+	name = "Hydrocarbon"
+	description = "A gunky mixture of hydrogen and carbon molecules, most often used in the production of aviation fuel."
+	reagent_state = LIQUID
+	color = "#e3f5f9"
+	taste_description = "oily water"
+
+/datum/reagent/aviation_fuel
+	name = "Tyrosene"
+	description = "High performance aviation fuel used in small fighter craft, created by enhancing standard fuel with extra hydrocarbons."
+	reagent_state = LIQUID
+	color = "#170B28"
+	taste_description = "jet fuel"
+
+
+/datum/looping_sound/refuel
+	mid_sounds = list('nsv13/sound/effects/fighters/refuel.ogg')
+	mid_length = 8 SECONDS
+	volume = 100
+
+/obj/structure/reagent_dispensers/fueltank/aviation_fuel
+	name = "Tyrosene fuel pump"
+	desc = "A tank full of high performance aviation fuel with an inbuilt fuel pump for rapid fuel delivery"
+	icon_state = "jetfuel" //NSV13 - Modified objects.dmi to include a jet fuel container
+	reagent_id = /datum/reagent/aviation_fuel
+	tank_volume = 3500
+	var/obj/item/jetfuel_nozzle/nozzle = null
+	var/obj/structure/overmap/fighter/fuel_target
+	var/datum/looping_sound/refuel/soundloop
+	var/max_range = 2
+	var/datum/beam/current_beam
+	var/mob/last_user
+	var/allow_refuel = FALSE
+
+/obj/structure/reagent_dispensers/fueltank/aviation_fuel/ui_act(action, params, datum/tgui/ui)
+	if(..())
+		return
+	if(!in_range(src, usr) && last_user != usr) //Topic check
+		return
+	if(action == "stopfuel" && fuel_target)
+		soundloop?.stop()
+		visible_message("<span class='warning'>[icon2html(src)] refuelling cancelled.</span>")
+		playsound(src, 'sound/machines/buzz-two.ogg', 100)
+		fuel_target = null
+		STOP_PROCESSING(SSobj, src)

```

Comment:
Do a automatic timer once connected to fuel up. We dont need a processing here I think.

Reply:
This way the fuelling can be done manually and precisely rather than it always filling from empty to full

Reply:
See airlocks it doesnt need from empty to full. There is a way toanage that without a processing I believe.

Reply:
ehhhh
This isn't costly at all and using something like a shedload of timers is just gonna mean the timers SS has more things to manage.
process() by itself isn't inherently bad, and this is selectively processing only when it needs to as opposed to constantly, while performing a very simple operation.

Reply:
I'm not sure airlocks are comparable, since you're transferring something over time instead of just performing one function on a delay. Things like cryopods and plumbing use process().